### ISSUE TITLE
Move off of Github

### ISSUE THREAD

**Issue by nothings** [nothings/stb#1353]
There are a number of reasons to consider moving off of github:

- (not a reason for this project: because stb is public domain, I'm not concerned about the copyright issues of Github Copilot)
- don't support Microsoft for political reasons (e.g. embrace / extend / extinguish)
- github's weird thing where repositories owned by "organizations" have more features than those owned by individuals, but the docs do not bother distinguishing the cases so you can waste a lot of time trying to figure out why what you're looking at doesn't match the screenshots in the documentation
- heavy-handed content moderation
- github not having clear explanation/descriptions of moderated content (and I mean technically, not the subjective policy part)

To address the last two, the issue I ran into is the reported post: https://github.com/nothings/stb/issues/359#issuecomment-1198716058

My best guess is that this content was reported to Github, and a Github moderator/bot looked at the post and marked it as "disruptive content", but it's impossible to know what that text means because googling for it just turns up a million similarly tagged posts, which is why I can only guess at the meaning. But I'm assuming this is the case Github describes in one of their docs as as them taking action by "downgrading the visibility of the offending content". So this is (a) moderation I don't appreciate triggered by a disruptive user having reported my post, and (b) terrible UX on Github's part at letting me understand what is even happening to the post.

**Comment by N-R-K** [comment]
We had to make a similar decision about moving away from github for the `nsxiv` project a couple months ago. [Here's a brief list of alternatives we considered and some of their pros and cons](https://codeberg.org/nsxiv/nsxiv-tmp/issues/1#issuecomment-465397).

At the end we decided on [Codeberg](https://codeberg.org) (which is powered by [Gitea](https://gitea.io/en-us/)) due to the following reasons:

* It's UI/UX is quite similar to github, so that results in less friction when moving for the developers.
* It supports "migrating" repos from github, meaning it'll "clone" all issues/PRs (but not "discussion" afaik) from an existing Github repo and create a new repo off that.
* Been around for 3+ years now and seem to be reliable as a service provider.

A couple problems you might face:

* Moving to a lesser known platform typically means less bug reports and/or contributions.
* A couple edges of the UI can be a bit unpolished at times.
* They do have a CI of their own, but it's currently at beta and limited-access. Though access is pretty much freely given within a day-ish without much issue. (There are also other third party CI providers which have support for gitea, so might not be an issue. Can't say for sure as I haven't used any 3rd party ones).

**Comment by onitake** [comment]
I respect your decision; while there's certainly value in having a central place to share code and collaborate, it creates a monoculture and that's rarely a good thing. Microsoft is partially to blame here, but Github is also somewhat a victim of its own success.

On the other hand, diverse code hosting services will reduce collaboration, as people need to be signed up to many different services, and private instances are often off limits, etc.
Perhaps this will eventually help bring people closer again: https://forgefed.org/

**Comment by nothings** [comment]
It's not a decision.

> There are a number of reasons to **consider** moving off of github:

**Comment by onitake** [comment]
> It's not a decision.
> 
> > There are a number of reasons to **consider** moving off of github:

Let me rephrase then - I would respect your decision if you decided to move off GH. :wink:

**Comment by N-R-K** [comment]
> On the other hand, diverse code hosting services will reduce collaboration, as people need to be signed up to many different services

Might I note that there exists [SourceHut](https://sourcehut.org/) which works mainly off of emails (or mailing list if you will).

* Want to submit a bug report? Just mail it to the relevant mailing list. No sign up required.
* Want to submit a patch? Same thing, just mail it.

I don't use sourcehut nor do I have an account on there, but I have contributed to a couple projects hosted there due to the simple mailing list workflow.

The downside, I presume, would be that developers not familiar with the mailing-list workflow will need to go through a bit of friction readjusting.
